### PR TITLE
MapView TimeExtent

### PR DESCRIPTION
**Pull request by shubham7109**
PR to add an `onTimeExtentChanged` and `timeExtent` properties to the MapView.

### CODE REVIEW

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt [hunk 1]
```diff
@@ -100,6 +104,7 @@ public fun MapView(
     onLongPress: ((LongPressEvent) -> Unit)? = null,
     onTwoPointerTap: ((TwoPointerTapEvent) -> Unit)? = null,
     onPan: ((PanChangeEvent) -> Unit)? = null,
+    onTimeExtentChanged: ((TimeExtent?) -> Unit)? = null,
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt:107]: I think we should move this parameter next to the `timeExtent` parameter. Alternatively, lets consider if a single `TimeExtentState` or `TimeAwareState` parameter would make sense here:
```kotlin
@Stable
data class TimeAwareState(
   val timeExtent: TimeExtent,
   val onTimeExtentChanged: ((TimeExtent?) -> Unit)? = null
)
```

Comment (shubham7109) [reply toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt]: Added the feedback from today's discussion to add the `timeExtent` next to the `onTimeExtentChanged` parameter.

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt [hunk 2]
```diff
@@ -178,9 +185,11 @@ private fun MapViewEventHandler(
     onLongPress: ((LongPressEvent) -> Unit)?,
     onTwoPointerTap: ((TwoPointerTapEvent) -> Unit)?,
     onPan: ((PanChangeEvent) -> Unit)?,
+    onTimeExtentChanged: ((TimeExtent?) -> Unit)?,
```
Comment (gunt0001) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt:188]: Can you move this up just above the `onViewpointChanged` parameter so it matches the parameter order in the `MapView()` function.

**Review by gunt0001 — APPROVED** [review]
@shubham7109 - looks good, one comment.

**Comment by shubham7109** [comment]
Thanks for the review @puneet-pdx, I've added the changes and updated the branch with the base feature branch. Ready for your re-review.

**Pull request merged**